src/plugins/data-card/data-card-tile.tsx

```diff
@@ -56,25 +56,17 @@ export const DataCardToolComponent: React.FC<ITileProps> = observer((props) => {
     });
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  useEffect(() => {
-    if (!tileElt) return;
-    const uiHeight = tileElt?.querySelector(".data-card-container")?.scrollHeight || 0;
-    const spaceLeft = height ? height - uiHeight : 0;
-    // no document id means we are in content, so we allow a change to the document model (row height)
-    if (!documentId && readOnly) onRequestRowHeight(model.id, Math.max(uiHeight, kExampleDeckHeight));
-    if (!readOnly) spaceLeft < kButtonSpace && onRequestRowHeight(model.id, uiHeight + kButtonSpace);
-    // first three dependencies are to pick up on content changes that may require a height change
-  }, [currEditAttrId, currEditFacet, imageUrlToAdd,
-    height, readOnly, tileElt, onRequestRowHeight, model.id, documentId]);
+  useDataCardTileHeight({
+    tileElt,
+    height: height && isFinite(height) ? height : 0,
+    currEditAttrId: currEditAttrId ?? "",
+    modelId: model.id,
+    documentId,
+    readOnly: readOnly ?? false,
+    onRequestRowHeight,
+    attrCount: attrIdsNames.length,
```

Comment:
I'd use `content.attributes.length` here instead. `attrIdsNames` will get recreated on each render because it is not a cached view.  Technically what you've got should work the same because the count won't change even if `attrIdsNames` does. But I think it would be more clear if it was using an object that doesn't get recreated each time.